Hfp 3539 bq add exit button

ravimajithia opened the pull request.

Commit:
HFP-3539 Fix BQ: not possible to tab out of the dialog

Commit:
HFP-3539 Fix support for spacebar

fnoks commented on branchingQuestion.js:
```diff
@@ -43,12 +43,66 @@ H5P.BranchingQuestion = (function () {
+        }
+      });
+
+      close.onclick = function (e) {
```
Please be consistent. I.e: use addEventListener('close', <...>) instead of onclick.

fnoks commented on branchingQuestion.js:
```diff
@@ -43,12 +43,66 @@ H5P.BranchingQuestion = (function () {
+    const closeDialog = function () {
+      const overlay = self.parent.libraryScreen.overlay;
+      if (overlay) {
+        // TODO: When does this code every run?!
```
This TODO should be resolved. I.e: I don't understand why this code was added if you don't know when it is executed?

ravimajithia replied:
Already improved, just forgot to remove the message.

fnoks commented on branchingQuestion.js:
```diff
@@ -43,12 +43,66 @@ H5P.BranchingQuestion = (function () {
+     * @return {Element} close
+     */
+    const createCloseElement = function () {
+      const close = document.createElement('div');
```
I believe we should try to use "button" instead. then we don't need to add special handling of the key events.

Commit:
HFP-3539 Improve code
- Make events consistant
- Focus was trapped in BQs now user can focus close button

fnoks committed:
HFP-3539 Fix redundant code and simplify
